### PR TITLE
Speed up Travis tests

### PR DESCRIPTION
**Pull request by mattrobenolt**
tl;dr this cuts the runtime by half on Travis.
- `SKIP_BACKEND_VALIDATION` since not really needed
- Don't install `codecov`, Travis has it pre-installed
- The JS linter was being run twice
- Parallelize the js and cli test suites in matrix build
- Each test suite only installs what is needed. `mysql` installs `mysql` stuff only, not `postgres`, for example.

By splitting out the js test suite, our python builds sqlite/mysql/postgres/cli now don't even attempt to touch `npm` or running webpack. And conversely, for js builds, there's no installing python dependencies.

I also removed the running of `webpack -p` due to the `postinstall` hook from `npm install` by doing `npm install --no-scripts`. The webpack build isn't necessary to tests to run, but not sure if it's nice to have their as a sanity check. My assumption was that eslint would have it covered. I'll happily add it back if found necessary.

/cc @dcramer @benvinegar

### CODE REVIEW

**Comment by dcramer** [comment]
Not a fan of codecov change since it's pretty magical to assume it exists and it should get cached

**Comment by mattrobenolt** [comment]
> Not a fan of codecov change since it's pretty magical to assume it exists and it should get cached

Trivial to add back, my point was that it's already pre-installed. So it doesn't need to be cached, and the `pip install` was just a noop.

**Comment by benvinegar** [comment]
🔥
